go/carmen/memory_footprint.go

```diff
@@ -0,0 +1,72 @@
+package carmen
+
+import "github.com/Fantom-foundation/Carmen/go/common"
+
+type MemoryFootprint interface {
+	// GetChild returns a child of the memory footprint with the given name.
+	GetChild(name string) MemoryFootprint
+
+	// AddChild allows to attach a new value under current memory footprint.
+	AddChild(name string, childValue uintptr)
```

Comment:
I guess we do not need (want) the client to modify the memory footprint obtained from Carmen, i.e. methods `AddChild` and `SetNote` should not be in the API